File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> st;
        int len = 0;

        for(auto n: nums) {
            st.insert(n);
        }

        for(int n: st) {
            int l = 0;
            if(st.find(n - 1) == st.end()) {

                while(st.find(n) != st.end()) {
                    l++;
                    n++;
                }
            }

            len = max(l, len);
        }

        return len;
    }
};

File: Data Structures & Algorithms/house-robber/submission-2.java
class Solution {
    public int rob(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        int n = nums.length;

        int n1 = rob(nums, map, n - 1);
        int n2 = rob(nums, map, n - 2);

        return Math.max(n1, n2);
    }

    public int rob(int[] nums, Map<Integer, Integer> map, int n) {
        if(map.containsKey(n)) return map.get(n);

        if(n == 0) return nums[0];

        if(n == 1) {
            return nums[1];
        }

        if(n < 0) {
            return 0;
        }

        int n1 = rob(nums, map, n - 2);
        int n2 = rob(nums, map, n - 3);

        map.put(n, nums[n] + Math.max(n1, n2));

        return map.get(n);
    }
}